controllers/pagerdutyintegration/serviceorchestration_handler.go

```diff
@@ -116,29 +115,33 @@ func (r *PagerDutyIntegrationReconciler) handleServiceOrchestration(pdclient pd.
 		return err
 	}
 
-	if pdiSelector.Matches(labels.Set(ruleConfigmap.GetLabels())) {
-		orchestrationConfig, err := utils.LoadConfigMapData(r.Client,
+	var orchestrationRuleConfigData string
+	if pdiSelector.Matches(labels.Set(orchestrationRuleConfigmap.GetLabels())) {
```

Comment:
The label selector checking is not added here, but in https://github.com/openshift/pagerduty-operator/blob/master/controllers/pagerdutyintegration/event_handlers.go#L231

The controller will watch for the configmaps which have the same label with the PDI.

It seems make sense to me to limit the configmap which match the labels to trigger the reconcile.

I will test if remove the if condition here, it will work the same.

Reply:
Tested with the `if pdiSelector.Matches(labels.Set(orchestrationRuleConfigmap.GetLabels()))` exists in `serviceorchestration_handler.go` or not.

When the if condition exists in `serviceorchestration_handler.go`
- When the configmap has the matched label, 
It will call the func `ApplyServiceOrchestration()` on new clusterdeployment creation and on service orchestration configmap updated.
- When the configmap does not have the matched label,
It will **not** call the func `ApplyServiceOrchestration()` on new clusterdeployment creation and on service orchestration configmap updated

When the if condition does not exist in `serviceorchestration_handler.go`
- When the configmap has the matched label, 
It will call the func `ApplyServiceOrchestration()` on new clusterdeployment creation and on service orchestration configmap updated.
- When the configmap does not have the matched label,
It will call the func `ApplyServiceOrchestration()` on new clusterdeployment creation but not on service orchestration configmap updated

Consider the label in the configmap has already defined/applied. 
The if logic in `serviceorchestration_handler.go` can be removed.